### PR TITLE
rc git-tools: Change the CWD before running some `git` commands

### PR DESCRIPTION
**Pull request by lenormf**
When opening buffers from a directory that's not under versioning
(or even from a different repository), `git` commands will not work.

This commit temporarily changes the current working directory to that
of the current buffer to make sure the context is right.

### CODE REVIEW

**Comment by lenormf** [comment]
Right, I fixed commands that cannot possibly be run from anywhere other than in the current buffer's repository.

File: rc/extra/git-tools.kak [hunk 1]
```diff
@@ -61,12 +69,13 @@ Available commands:\n  add\n  rm\n  blame\n  commit\n  checkout\n  diff\n  hide-
...
 
     run_git_blame() {
+        cd_bufdir
         (
             printf %s "evaluate-commands -client '$kak_client' %{
                       try %{ add-highlighter window/git-blame flag-lines GitBlame git_blame_flags }
-                      set-option buffer=$kak_bufname git_blame_flags '$kak_timestamp'
-                  }" | kak -p ${kak_session}
-                  git blame "$@" --incremental ${kak_buffile} | awk '
+                      set-option buffer='${kak_bufname}' git_blame_flags '$kak_timestamp'
```
[the remark this answers is not in the document]
Comment (lenormf) [reply rc/extra/git-tools.kak]: Oops, been doing too much shell!

**Comment by lenormf** [comment]
Bump!

**Comment by ath3** [comment]
Wouldnt it be a good solution to have a bool option if cd_bufdir() should be called or not?
That way the users can set it to their preference.

**Pull request merged**